soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -3888,6 +3901,33 @@ void DrawRandoEditor(bool& open) {
                 ImGui::BeginChild("ChildMiscWorldSettings", ImVec2(0,-8));
                 ImGui::PushItemWidth(-FLT_MIN);
 
+                //Starting Age
+                //Disabled when Forest is set to Closed
+                bool disableRandoStartingAge = !CVar_GetS32("gRandomizeForest", 0);
+                const char* disableRandoStartingAgeText = "This option is disabled because \"Forest\" is set to \"Closed\"";
+                ImGui::Text(Settings::StartingAge.GetName().c_str());
+                if (!disableRandoStartingAge) {
+                UIWidgets::InsertHelpHoverText(
+                    "Choose which age Link will start as.\n\n"
+                    "Starting as adult means you start with the Master Sword in your inventory.\n"
+                    "Only the child option is compatible with Closed Forest."    
+                );
+                    UIWidgets::EnhancementCombobox("gRandomizeStartingAge", randoStartingAge, 3, 0);
+                }
+                else {
+                    ImGui::PushItemFlag(ImGuiItemFlags_Disabled, true);
+                    ImGui::PushStyleVar(ImGuiStyleVar_Alpha, ImGui::GetStyle().Alpha * 0.5f);
+                    if (ImGui::IsItemHovered(ImGuiHoveredFlags_AllowWhenDisabled) && strcmp(disableRandoStartingAgeText, "") != 0) {
+                       ImGui::SetTooltip("%s", disableRandoStartingAgeText);
+                    }
+                    CVar_SetS32("gRandomizeStartingAge", 1);
+                    UIWidgets::EnhancementCombobox("gRandomizeStartingAge", randoStartingAge, 3, 1);
+                    ImGui::PopStyleVar(1);
+                    ImGui::PopItemFlag();
+                }
```

Comment:
I feel like this should be possible without as much copy-pasta. Couldn't we wrap only the neccesary elements in an if statement, and use the same dropdown for wether it's disabled or not disabled? So you'd wrap the ImGui flag and stylevar stuff in an if statement checking if it's disabled above and below the combobox.

I also think the strcmp in there is unneccesary, as there will always be text to show in the first place.